src/Development/IDE/Spans/Calculate.hs

```diff
@@ -158,15 +158,39 @@ getTypeLHsExpr tms e = do
     Nothing -> return Nothing
   where
     getSpanSource :: HsExpr GhcTc -> SpanSource
-    getSpanSource (HsLit U lit) = Lit (showGhc lit)
-    getSpanSource (HsOverLit U lit) = Lit (showGhc lit)
+    getSpanSource xpr | isLit xpr = Lit (showGhc xpr)
     getSpanSource (HsVar U (L _ i)) = Named (getName i)
     getSpanSource (HsConLikeOut U (RealDataCon dc)) = Named (dataConName dc)
     getSpanSource RecordCon {rcon_con_name} = Named (getName rcon_con_name)
     getSpanSource (HsWrap U _ xpr) = getSpanSource xpr
     getSpanSource (HsPar U xpr) = getSpanSource (unLoc xpr)
     getSpanSource _ = NoSource
 
+    isLit :: HsExpr GhcTc -> Bool
+    isLit (HsLit U _)     = True
+    isLit (HsOverLit U _) = True
+    isLit (ExplicitTuple U args _) = all (isTupLit . unLoc) args
+      where
+        isTupLit (Present U xpr) = isLit (unLoc xpr)
+-- the following #if is there
+-- to please HLint in all versions
+#if MIN_GHC_API_VERSION(8,6,0)
+        isTupLit (Missing U)     = True
```

Comment:
Treating tuple sections as literals feels a bit weird. Is this really what we want here?

Reply:
Right, this was a mistake.